aws-storage-s3/src/main/java/com/amplifyframework/storage/s3/operation/AWSS3StorageListOperation.java

```diff
@@ -64,35 +65,21 @@ public AWSS3StorageListOperation(
     @Override
     public void start() {
         executorService.submit(() -> {
-            String identityId;
-
             try {
-                identityId = AWSMobileClient.getInstance().getIdentityId();
-
-                try {
-                    StorageListResult result = storageService.listFiles(
-                        S3RequestUtils.getServiceKey(
-                            getRequest().getAccessLevel(),
-                            identityId,
-                            getRequest().getPath(),
-                            getRequest().getTargetIdentityId()
-                        )
-                    );
+                List<StorageItem> listedItems = storageService.listFiles(
+                    S3RequestUtils.getServiceKey(
+                        getRequest().getAccessLevel(),
+                        getRequest().getTargetIdentityId(),
+                        getRequest().getPath()
+                    )
+                );
 
-                    onSuccess.accept(result);
-                } catch (Exception exception) {
-                    onError.accept(new StorageException(
-                        "Something went wrong with your AWS S3 Storage list operation",
-                        exception,
-                        "See attached exception for more information and suggestions"
-                    ));
-                }
+                onSuccess.accept(StorageListResult.fromItems(listedItems));
             } catch (Exception exception) {
```

Comment:
Any change to `catch (SpecificThingOne fineGrainedCatch)`, `catch (SpecificThingTwo fineGrainedThignTwo)` ? Will help create a less generic error message in the `StorageException` that you create